runners/spark/src/test/java/org/apache/beam/runners/spark/translation/streaming/SparkCoGroupByKeyStreamingTest.java

```diff
@@ -0,0 +1,150 @@
+  @Test
+  public void testInStreamingMode() throws Exception {
+    Instant instant = new Instant(0);
+    CreateStream<KV<Integer, Integer>> source1 =
+        CreateStream.of(KvCoder.of(VarIntCoder.of(), VarIntCoder.of()), batchDuration())
+            .emptyBatch()
+            .advanceWatermarkForNextBatch(instant)
+            .nextBatch(
+                TimestampedValue.of(KV.of(1, 1), instant),
+                TimestampedValue.of(KV.of(1, 2), instant),
+                TimestampedValue.of(KV.of(1, 3), instant))
+            .advanceWatermarkForNextBatch(instant.plus(Duration.standardSeconds(1L)))
+            .nextBatch(
+                TimestampedValue.of(KV.of(2, 4), instant.plus(Duration.standardSeconds(1L))),
+                TimestampedValue.of(KV.of(2, 5), instant.plus(Duration.standardSeconds(1L))),
+                TimestampedValue.of(KV.of(2, 6), instant.plus(Duration.standardSeconds(1L))))
+            .advanceNextBatchWatermarkToInfinity();
+
+    CreateStream<KV<Integer, Integer>> source2 =
+        CreateStream.of(KvCoder.of(VarIntCoder.of(), VarIntCoder.of()), batchDuration())
+            .emptyBatch()
+            .advanceWatermarkForNextBatch(instant)
+            .nextBatch(
+                TimestampedValue.of(KV.of(1, 11), instant),
+                TimestampedValue.of(KV.of(1, 12), instant),
+                TimestampedValue.of(KV.of(1, 13), instant))
+            .advanceWatermarkForNextBatch(instant.plus(Duration.standardSeconds(1L)))
+            .nextBatch(
+                TimestampedValue.of(KV.of(2, 14), instant.plus(Duration.standardSeconds(1L))),
+                TimestampedValue.of(KV.of(2, 15), instant.plus(Duration.standardSeconds(1L))),
+                TimestampedValue.of(KV.of(2, 16), instant.plus(Duration.standardSeconds(1L))))
+            .advanceNextBatchWatermarkToInfinity();
+
+    PCollection<KV<Integer, Integer>> input1 =
+        pipeline
+            .apply("create source1", source1)
+            .apply(
+                "window input1",
+                Window.<KV<Integer, Integer>>into(FixedWindows.of(Duration.standardSeconds(3L)))
+                    .withAllowedLateness(Duration.ZERO));
+
+    PCollection<KV<Integer, Integer>> input2 =
+        pipeline
+            .apply("create source2", source2)
+            .apply(
+                "window input2",
+                Window.<KV<Integer, Integer>>into(FixedWindows.of(Duration.standardSeconds(3L)))
+                    .withAllowedLateness(Duration.ZERO));
+
+    PCollection<KV<Integer, CoGbkResult>> output =
+        KeyedPCollectionTuple.of(INPUT1_TAG, input1)
+            .and(INPUT2_TAG, input2)
+            .apply(CoGroupByKey.create());
+    PAssert.that("wrong output of the join using CoGroupByKey in streaming mode", output)
+        .satisfies(
+            (SerializableFunction<Iterable<KV<Integer, CoGbkResult>>, Void>)
+                input -> {
+                  for (KV<Integer, CoGbkResult> element : input) {
```

Comment:
Does this assert fail if `input` has 0 or more than 2 entries? 
Another potential corner case when keys (for unknown reason) have different values than expected (1 and 2).
In these both cases we just don't get inside `if` statements, it will be skipped and test passes, no?